Add review_checklist.md (#127)
ansible-collections/community.mysql: review_checklist.md

# Review Checklist

When reviewing, keep in mind that we follow [Ansible Code of Conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html) in all our contributions and interactions within this repository.

**General tips**

> - Try create a culture of collaboration, when reviewing

Reviewer: I'm not sure about this change myself though, but I think "try create" is not correct.
```suggestion
- Try to create a culture of collaboration when reviewing
```

Author: Good spot, thanks! committed

- Welcome the author and thank them for the pull request
- When suggesting changes, try to use questions, not statements
- When suggesting mandatory changes, do it as politely as possible providing documentation references
- If your suggestion is optional or a matter of personal preferences, please say it explicitly
- When asking for adding tests or for complex code refactoring, say that the author is welcome to ask for clarifications and help if they need
- If somebody suggests a good idea, say about it or put a thumbs up
- After merging, thank the author and reviewers for their time and effort

**Standards and documentation**

> - [ ] if the pull request is not a documentation fix, it must include a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#creating-a-changelog-fragment). Please check the format carefully

Reviewer: ```suggestion
- [ ] if the pull request is not a documentation fix, it must include a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#creating-a-changelog-fragment). Please check the format carefully.
```

Author: yes, i know about it. If I add the dot, not sure if the next sentence should start with a capital letter and so on.. let's try another approach

Author: PTAL

Reviewer: Thanks for the change, it's more convenient now imho.

Author: @aminvakil thanks for spotting this and giving the feedback

- [ ] if new files are added with the pull request, they follow the [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
- [ ] the changes follow the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [ ] follow [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)

> - [ ] if a new plugin is added, it is one of the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#id14)

Reviewer: ```suggestion
- [ ] if a new plugin is added, it is one of the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins)
```

- [ ] documentation, examples, and return sections use FQCNs for `M(..)` when referencing to modules
- [ ] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix when mentioned
- [ ] when a new option, module, plugin, or return value is added, the corresponding documentation or return sections use `version_added:` containing the *collection* version for which they were added
- [ ] FQCNs is used for `extends_documentation_fragment:`, unless the author is referring to doc_fragments from ansible-core

**Tests (if applicable and possible to implement):** 
- [ ] the pull request has integration tests 
- [ ] the pull request has unit tests
- [ ] all changes are covered

> - [ ] integration tests also cover `check_mode`

Reviewer: ```suggestion
- [ ] integration tests also cover `check_mode` (if it is supported)
```

Author: well spotted, committed:)
